### ISSUE TITLE
HUD CSV Export version FY2020 from SP: waiting on corrections from WS

### ISSUE THREAD

**Issue by kiadso** [COHHIO/COHHIO_HMIS#12]
The HUD CSV Export from ServicePoint is much improved over the past year, but some minor defects remain. Until they're fixed, be aware that the code in all of my repositories is written to work around these problems, and as they are fixed, the code will be adjusted.

WellSky Ticket | The Problem | What I'm Doing While it's being Fixed
---|---| ---
864582 | Yes/No data (e.g. IncomeFromAnySource) differs from SP data bc WS is overwriting nulls with whatever the subs are | WS will fix this. Since we're missing false negatives, at least it's not causing users to go hunt for something that's not a problem.
864459 | LoSUnderThreshold should have 0, 1, or 99, but instead it has True/False data. | The HUD CSV specs say this data should return either a 0, a 1, a 99, or a null. The reason this matters is I can't tell if a null means it was irrelevant based on the other answers or if it's null because it was not answered. Asked them to re-open the case on 11/12/2019. Currently it's coded around.
877960 | Last Permanent Address: State field not populating | submitted to DEV 10/18/2019
886040 | CurrentLivingSituation.csv has some nulls in the CLS column | not using this yet anyway
886055 | Users.csv only includes current users | should be fixed because UserIDs are referred to throughout the other data objects, and will not have anything to tie back to. Using ART report anyway bc I also need Default Provider and that's the only way to get it.
807291 | the export is double quoted | This only affects when I use the export to upload our HIC data to HDX. HDX rejects it. The fix for this (from my side) is located in the script in the HIC repository (https://github.com/COHHIO/HIC) until they fix it on their side.
~873163~ | ~Export data not matching what is in SP, specifically with Residence Prior and Length of Stay~ | ~submitted to DEV 9/30. Not fixed yet BUT they did find the reason. They didn't map old answers to one of the LoTH questions correctly, so the old answers are just coming in as nulls. I asked them to add "(retired)" to the value so users can tell what's wrong, but they closed the case.~
~918016~ | ~Organization ID in Project.csv shouldn't have nulls~ | ~submitted ticket 2/14/2020. No need to adjust anything in my code, it hasn't affected my work (yet)~
~886053~ | ~InformationDate column in CurrentLivingSituation.csv formatted incorrectly~ | ~commented out that line of code so I can use it once it's fixed, copied it and "corrected" it so it works with the wrongness of this export. shouldn't affect anything I need currently.~
~886038~ | ~HMIS Participating in Projects.csv is all 1's~ | UPDATE: this has been fixed. ~ORIGINAL: put in a case, overwriting data from ReportWriter~
~837944~ | ~subassessments are coming in based on their Start & End Dates as compared to the Report Start & End Dates instead of the Entry & Exit Dates of the Enrollment ID that they're attached to~ | UPDATE: fixed 2/13/2020 ~filtering out any Data Quality issues to do with subassessments, being sure the Data Quality reports have "UNDER CONSTRUCTION" written in the header until this is corrected~
~858797~ | ~2 identical Effective Dates & different Values returns the wrong value in the export~ | UPDATE: This has been fixed. ~ORIGINAL COMMENT: this is the 3rd ticket about this issue. I have tried (again) to impress upon them that this is an export-wide thing they need to fix but they seem to be relying on me for examples of Client IDs where this problem is evident. Trying to escalate as much as possible.~
~~833259~~ | ~~takes 12 hours to run 2 years of data~~ | ~~running the export prior to leaving work, downloading next morning.~~
~~804998~~ | ~~Incorrect Enrollment User ID. The User ID in the Enrollment file and in the Client file is actually the Project ID.~~ | ~~Importing this data with RMisc (created in ART) and replacing the bad data with data from ART.~~
~~810738~~ | ~~No Bed Types are coming through~~ | ~~Pulling it in from ReportWriter until WS fixes it.~~
~~818626~~ | ~~No Provider Addresses coming through~~ | ~~Pulling it in from the RMisc file until WS fixes it.~~ Resolved 10/30
~~818854~~ | ~~No Youth Beds coming through~~ | ~~Pulling it in from ReportWriter until WS fixes it.~~
~~873523~~ | ~~Move-In Date in Export returning Move-In Date of the HoH on non-HoHs even if that non-HoH hadn't joined the project stay yet.~~ | See comments below for more on this one.
~~874938~~ | ~~FY2020 HUD CSV Export still not available~~ | ~~??? It was due October 1, it's October 23rd and still no updated export. The next promise is October 28th.~~ Update: October 30th, it's available.

**Comment by kiadso** [comment]
Update for Ticket 833259: This is done. It now takes 45 minutes to run two years of data on our site.

**Comment by kiadso** [comment]
Added a new issue 864459. LOSUnderThreshold seems off? Not sure, though.

**Comment by kiadso** [comment]
Added another issue 864582. Health Insurance y/n is coming out incorrectly?? Hopefully I'm just missing something??

**Comment by kiadso** [comment]
Case 864582 applies to all the yes/no data elements with subassessments involved. WS is not yet clear on whether or not they're going to fix the problem.

**Comment by kiadso** [comment]
Found another thing, where the data in SP is not matching what's in the Export. (Added a row for the new case.)

**Comment by kiadso** [comment]
I'm hearing from WS that they're going to fix 6.12 and put the FY2020 one out at the same time. (October 1)

**Comment by kiadso** [comment]
**Regarding 873523**: WS is overwriting the Move-In Date of non-HoHs with the Move In Date of the HoH if they have different Entry Dates than the HoH. I understand why they did this but because the Enrollment.csv is organized the way it is, plus the fact that Move In Date is supposed to be Household-level only (even though there are PersonalIDs in the Enrollment.csv), I decided to code the Move In Date Adjust to replace the non-HoH's Move In Date with their Entry Date if their Entry Date doesn't match the HoH's Entry Date. Discussed with Mary and she said this makes sense for my local purposes. Will put in an AAQ suggesting the HUD CSV specs be corrected so that there are no Personal IDs in a file that should only be at a household level.

**Comment by kiadso** [comment]
The unavailability of the HUD CSV Export FY2020 continues to present a giant problem in what I'm trying to do. CW assures me that by this coming Monday at the latest it will be out and she's going to contact me on Friday for an update.

**Comment by kiadso** [comment]
FY2020 finally came available. Unknown which of my tickets were addressed. WS is working on getting this information for me. For sure the Provider addresses were addressed.

**Comment by kiadso** [comment]
Initiated weekly calls with the ppl deciding about when to put things into sprints and I'll try to update what I find out here. Today I found out 873163 is in the current sprint and will roll over into the next ones til it's done.

The following cases are in the next sprint which starts Monday the 25th: 858797, 886053, 886040.

837944 will be in the sprint after that (December 9th)

Apparently sprints are 2 weeks apart.

864459- she hasn't created a Jira ticket but she will

I didn't press on the others. The most important one is 858797, next important is 837944, CW knows that.

**Comment by kiadso** [comment]
The last meeting was 12/6. 
Breakthrough on 858757, should be completed by the end of the month because testing will be more difficult than coding the fix.
837944 was not in any sprint, but I asked her to trade it for another ticket because it's high priority.

**Comment by kiadso** [comment]
Twice a month calls with LH and CW have helped keep things moving, so I just knocked out a bunch of cases and moved them to the bottom. In my last communications, I asked that the top two be prioritized: 864582 and 864459 but I don't have an answer.

**Comment by kiadso** [comment]
Closing this issue since most of the major issues have been resolved. The ones left are minor to me, but may not be minor to others. I'm going to stop updating this.
